### DISCUSSION TITLE
Add the number of non zero coefficients to the solver interface

### DISCUSSION THREAD

**Discussion by amn14** [google/or-tools#4321] [Linear Solver questions]
**What language and solver does this apply to?**
All, Python, Java, C#,  Linear Solver

**Describe the problem you are trying to solve.**
Is it possible to add a function that get the count of non zero coefficients ? 
**Describe the solution you'd like**
In current version, there is no way to get directly the number of non zero coefficients. The developer has to maintain it, it would be great to have it directly in the solver interface.
**Describe alternatives you've considered**
Implementing it :

     static int CountNonZeroCoeffs(Solver solver)
     {
         int nonZeroCount = 0;
    
         var objective = solver.Objective();
         foreach (var variable in solver.variables())
         {
             foreach (var constraint in solver.constraints())
             {
                 if (constraint.GetCoefficient(variable) != 0)
                     nonZeroCount++;
             }
             if (objective.GetCoefficient(variable) != 0 )
                 nonZeroCount++;
         }
         return nonZeroCount;
     }

**Comment by lperron** [comment]
So, no, we do not maintain it. 
And you are creating the model, you can maintain it.

**Reply by amn14** [reply]
Thanks for your quick reply. 
Since you maintain the number of variables and constraints, I imagined it could be the same for non zero coeffs but I guess it's not the same logic (I didn't dig into the details of the code).
But as you said, it's not difficult to maintain it for the model owner too.
